tiledb/api/src/query/read/callback.rs

```diff
@@ -497,6 +526,13 @@ macro_rules! query_read_callback {
                     }
                 }
             }
+
+            impl<'ctx, 'data, T, B> ReadQueryBuilder<'ctx, 'data> for $Builder<'data, T, B>
```

Comment:
(This is inside the macro for defining the static arg read callback stuff)

TODO: use the `VarRawReadQuery` as a basis for those, as that will repeat less code.